enterprise/server/cmd/server/main.go

```diff
@@ -302,5 +303,7 @@ func main() {
 		log.Fatalf("%v", err)
 	}
 
+	realEnv.SetSingleFlightDeduper(dsingleflight.New(realEnv.GetDefaultRedisClient()))
```

Comment:
redis is optional so you may want to make this conditional. we typically do this by adding a `Register(env)` function in the component itself.

Reply:
oh good call, fixed